turbinia/workers/file_system_timeline.py

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+# Copyright 2022 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Task to run dfimagetools FileEntryLister on disk partitions."""
+from __future__ import unicode_literals
+
+import os
+
+from turbinia import TurbiniaException
+from turbinia.workers import TurbiniaTask
+from turbinia.evidence import EvidenceState as state
+from turbinia.evidence import BodyFile
+
+if TurbiniaTask.check_worker_role():
+  try:
+    from dfvfs.helpers import volume_scanner
+    from dfimagetools import file_entry_lister
+  except ImportError as exception:
+    message = 'Could not import libraries: {0!s}'.format(exception)
+    raise TurbiniaException(message)
+
+
+class FileSystemTimelineTask(TurbiniaTask):
+
+  REQUIRED_STATES = [state.ATTACHED]
+
+  def run(self, evidence, result):
+    """Task to execute (dfimagetools) FileEntryLister.
+
+    Args:
+        evidence (Evidence object):  The evidence we will process.
+        result (TurbiniaTaskResult): The object to place task results into.
+
+    Returns:
+        TurbiniaTaskResult object.
+    """
+    bodyfile_output = os.path.join(self.output_dir, 'file_system.bodyfile')
+    output_evidence = BodyFile(source_path=bodyfile_output)
+    output_evidence.parent_evidence = evidence
```

Comment:
I don't think you need to set the parent evidence here as that's mostly for  compound evidence types (e.g. GoogleCloudDisk) that need multiple pre-processors to run.  I just saw that the docstrings for this aren't very clear though, so I updated this issue to mention also adding some better comments for that (https://github.com/google/turbinia/issues/598).

Reply:
That makes sense. I wasn't sure what the actual purpose was so I set it because this generates a BodyFile evidence type which is used as input for Plaso. I will remove that line on the next commit.